寫一個程式，指定4個變數值為 10, 10.01, 10+10j與DICE，分別印出各自的資料型態。
使用type可以印出資料型態，可以參照程示範例。

輸入範例:


輸出範例:
<class 'int'>
<class 'float'>
<class 'complex'>
<class 'str'>

程示範例：以印出num1的型態為例
num1 = 10
num2 = 10.01
num3 = 10+10j
text='DICE'
print(type(num1))

In [1]:
'''
4_1.浮點數與其輸出格式-資料型態(15.0)
'''
variableA = 10
variableB = 10.01
variableC = 10+10j
variableD = "DICE"

mylist= [variableA,variableB,variableC,variableD]


for i in mylist:
    print(type(i))
    
    


<class 'int'>
<class 'float'>
<class 'complex'>
<class 'str'>


寫一個程式，指定二個變數，設定其值為 10.1 與 5.2，並且印出結果。
使用{0}來印出輸出結果，請參閱程示範例。

輸入範例:

輸出範例: 
num1=10.1
num2=5.2

程示範例：以第一個浮點數為例
num1 = 10.1
print("num1={0}".format(num1))

In [2]:
'''
4_2.浮點數與其輸出格式-印出浮點數-1(15.0)
'''

num1=10.1
num2=5.2

mylist=[num1,num2]

print("num1={0}\nnum2={1}".format(mylist[0],mylist[1]))

num1=10.1
num2=5.2


寫一個程式，指定兩個變數，分別為浮點數 10.1 與 5.2，並且印出結果。
使用{0:f}來印出輸出結果，請參閱程示範例。

輸入範例:

輸出範例: 
num1=10.100000
num2=5.200000

程示範例：以第一個浮點數為，例印出小數點後6位
num1 = 10.1
print("num1={0:f}".format(num1))

In [3]:
'''
4_3.浮點數與其輸出格式-印出浮點數-2(15.0)
'''

num1=10.1
num2=5.2

print("num1={0:f}".format(num1))
print("num2={0:f}".format(num2))


num1=10.100000
num2=5.200000


寫一個程式，指定二個變數，分別為浮點數 10.1 與 5.2，並且印出結果，分別印出小數點後2位與後3位。

輸入範例:

輸出範例: 
num1=10.10
num2=5.200

程示範例：以第一個浮點數為例，印出小數點後2位
num1 = 10.1
print("num1={0:.2f}".format(num1))

In [4]:
'''
4_4.浮點數與其輸出格式-印出浮點數-3(15.0)
'''

num1=10.10
num2=5.200

print("num1={0:.2f}".format(num1))
print("num2={0:.3f}".format(num2))

num1=10.10
num2=5.200




寫一個程式，指定二個浮點數為 10.1 與 5.2，並且印出結果。
分別印出小數點後2位與後3位。

輸入範例:

輸出範例: 
num1=10.10 num2=5.200

程示範例：以第一個浮點數為例
num1 = 10.1
num2 = 5.2
print("num1={0:.2f} num2={1:.3f}".format(num1, num2))

In [5]:
'''
4_5.浮點數與其輸出格式-印出浮點數-4(15.0)
'''
num1=10.1
num2=5.2


print("num1={0:.2f} num2={1:.3f}".format(num1,num2))

num1=10.10 num2=5.200


寫一個程式，設定兩個變數，其值分別為160與55.22，分別代表身高與體重。輸出時以浮點數印出其值，小數位數請印出兩位數，總數字寬度印出 10 位。

輸入範例:

輸出範例: 
My height is        160.00 CM.
My weight is         55.22 KG.

程示範例：以身高為例
height= 160
print('My height is {0:10.2f} CM.'.format(身高的變數))

In [6]:
'''
4_6.浮點數與其輸出格式-固定格式(10.0)
'''
variableA=160
variableB=55.22

print("My height is{0:10.2f} CM.".format(variableA))
print("My weight is{0:10.2f} KG.".format(variableB))

My height is    160.00 CM.
My weight is     55.22 KG.


寫一個程式，指定一個變數，其浮點數值為 3.141592653589793，然後執行下列任務。

以小數點兩位格式印出
再將此數字轉為整數，以變數名稱是num為例，轉型的寫法(以變數num為例)：
num = int(num)
印出整數
最終的數字型態，以變數名稱是num為例，印出變數型態的寫法print(type(num))
輸入範例:

輸出範例: 
3.14
3
<class 'int'>

In [7]:
'''
4_7.浮點數與其輸出格式-浮點數轉整數(15.0)
'''
pi= 3.141592653589793

print(f"{pi:.2f}")
pi=int(pi)
print(int(pi))
print(type(pi))

3.14
3
<class 'int'>


寫一個程式，指定一個字串變數為 KNQXG[QW，將其十進位值減2，就會知道我的心意。

輸入範例:
 
輸出範例:
ILOVEYOU

In [8]:
'''
4_8.浮點數與其輸出格式-自主學習(5.0)
'''
message = "KNQXG[QW"

decoded_message = "".join([chr(ord(char)-2)for char in message])

print(decoded_message)

ILOVEYOU
